WordPress/Classes/ViewRelated/Blog/QuickStartTourGuide.swift

```diff
@@ -126,10 +135,26 @@ open class QuickStartTourGuide: NSObject {
             fallthrough
         default:
             currentTourState = TourState(tour: tour, blog: blog, step: 0)
-            showCurrentStep()
+
+            if immediately {
+                showCurrentStep()
+            }
         }
     }
 
+    /// Runs the current tour.  Only works after calling `startTour` with parameter `immediately` set to
+    /// `false`, and if the tour is in the first step.  Otherwise this method will do nothing.
+    ///
+    func run() {
```

Comment:
`run` is a pretty ambiguous name, and doesn't fit the verbs used elsewhere in this class.

What about putting this guard at the start of `showCurrentStep` instead?

Reply:
I considered using `initialize()` and `start()`, thus removing the `immediate` param and forcing both to always be called (immediately or not, it'd be up to the caller).  Maybe that's a better option?  Any suggestion to make the method names clearer is of course welcome!

As to moving the guard in this method into `showCurrentStep()`, I'm not sure I understand: these methods have different purposes (`showCurrentStep()` should be called for all steps, whereas `run()` should only be called to run the tutorial from the first step forward.  But maybe this point would become less important if we go for something like `initialize` / `start`.

Reply:
Maybe `prepare` and `start`?

Reply:
Yeah, I liked those names. I prefer `begin` to `start` but it's a minor difference—start sounds immediate to me.

Reply:
Changed to `prepare()` and `begin()`.